doc/man3/EVP_PKEY_new.pod

```diff
@@ -168,7 +168,18 @@ general private key without reference to any particular algorithm.
 The structure returned by EVP_PKEY_new() is empty. To add a private or public
 key to this empty structure use the appropriate functions described in
 L<EVP_PKEY_set1_RSA(3)>, L<EVP_PKEY_set1_DSA(3)>, L<EVP_PKEY_set1_DH(3)> or
-L<EVP_PKEY_set1_EC_KEY(3)>.
+L<EVP_PKEY_set1_EC_KEY(3)> for legacy key types or for fully provider-managed
+key types (see L<provider-keymgmt(7)>), functions such as
+L<EVP_PKEY_set1_encoded_public_key(3)>.
+
+Generally a user of an B<EVP_PKEY> structure should not rely on it being
+valid across different library contexts. Under specific circumstances, though,
+it is possible that an B<EVP_PKEY> structure can be used in a library context B
+after having been created in a library context A, namely iff both library
```

Comment:
Nit: iff - is it intentional (if and only if) or just a typo?

Reply:
the former

Reply:
I think there are other circumstances where it would work. For example legacy/engine-based EVP_PKEYs should be also shareable because engines are not libctx-specific.

Also this sounds like this situation is exceptional however this import/export will be implemented and will work in most cases. The exception would be unexportable private keys or third party providers not implementing the export function at all. They have to implement import to support EVP_PKEY_fromdata so providers not implementing import would be very rare.

Reply:
Well, the un-exportable private keys are one of the reasons why I formulated things so "conservatively": What if providers don't agree on the same "exported" key format (or simply cannot parse a specific export format)? With ML-KEM we already run the risk of this happening if default provider would follow (only) IETF and oqsprovider keeps following (only) NIST.

But then again, that can also be formulated. Let me try differently in an update -- and hope people appreciate this precision (that's IMO pretty hard to ascertain as a user).